datahub-web-react/src/app/entity/user/UserEditProfileModal.tsx

```diff
@@ -0,0 +1,242 @@
+import { message, Button, Input, Modal, Typography, Form } from 'antd';
+import { useUpdateCorpUserPropertiesMutation } from '../../../graphql/user.generated';
+
+type Props = {
+    visible: boolean;
+    onClose: () => void;
+    onCreate: () => void;
+    data: {
+        name: string | undefined;
+        title: string | undefined;
+        image: string | undefined;
+        team: string | undefined;
+        email: string | undefined;
+        slack: string | undefined;
+        phone: string | undefined;
+        urn: string | undefined;
+    };
+};
+/** Regex Validations */
+export const validUserName = new RegExp('^[a-zA-Z ]*$');
+
+export default function UserEditProfileModal({ visible, onClose, onCreate, data }: Props) {
+    const [updateCorpUserPropertiesMutation] = useUpdateCorpUserPropertiesMutation();
+    const [form] = Form.useForm();
+
+    const [userName, setUserName] = useState<string | undefined>('');
+    const [userTitle, setUserTitle] = useState<string | undefined>('');
+    const [userImageURL, setImageURL] = useState<string | undefined>('');
+    const [userTeam, setUserTeam] = useState<string | undefined>('');
+    const [userEmail, setUserEmail] = useState<string | undefined>('');
+    const [userSlack, setUserSlack] = useState<string | undefined>('');
+    const [userPhoneNumber, setUserPhoneNumber] = useState<string | undefined>('');
+    const [buttonDisabled, setButtonDisabled] = useState(true);
+
+    useEffect(() => {
+        if (data) {
+            setUserName(data.name);
+            setUserTitle(data.title);
+            setImageURL(data.image);
+            setUserTeam(data.team);
+            setUserEmail(data.email);
+            setUserSlack(data.slack);
+            setUserPhoneNumber(data.phone);
+        }
+    }, [data]);
+
+    // save changes function
+    const onCreateGroup = () => {
+        updateCorpUserPropertiesMutation({
+            variables: {
+                urn: data?.urn || '',
+                input: {
+                    displayName: userName,
+                    title: userTitle,
+                    pictureLink: userImageURL,
+                    teams: userTeam?.split(','),
+                    email: userEmail,
+                    slack: userSlack,
+                    phone: userPhoneNumber,
+                },
+            },
+        })
+            .catch((e) => {
+                message.destroy();
+                message.error({ content: `Failed to Save changes!: \n ${e.message || ''}`, duration: 3 });
+            })
+            .finally(() => {
+                message.success({
+                    content: `Changes saved.`,
+                    duration: 3,
+                });
+                onCreate();
+                setUserName('');
+                setUserTitle('');
+                setImageURL('');
+                setUserTeam('');
+                setUserEmail('');
+                setUserSlack('');
+                setUserPhoneNumber('');
```

Comment:
why is this step needed? This seems like it would make it appear that the save failed

Reply:
We are assigning the blank values once the form is submitted.

Reply:
If we had a single object to represent all state, this would look cleaner!

Reply:
Sure, required changes are added